storage/reads/flux_reader.go

```diff
@@ -8,14 +8,43 @@ import (
 	"github.com/gogo/protobuf/types"
```

Comment:
```
$ git diff master:./storage/flux/reader.go ./storage/reads/flux_reader.go
diff --git a/storage/flux/reader.go b/storage/reads/flux_reader.go
index e02cabc304..b69eb290ac 100644
--- a/storage/flux/reader.go
+++ b/storage/reads/flux_reader.go
@@ -1,12 +1,10 @@
-package storageflux
+package reads
 
 import (
 	"context"
 	"fmt"
 	"strings"
 
-	"github.com/influxdata/influxdb/v2/kit/platform/errors"
-
 	"github.com/gogo/protobuf/types"
 	"github.com/influxdata/flux"
 	"github.com/influxdata/flux/execute"
@@ -14,11 +12,11 @@ import (
 	"github.com/influxdata/flux/memory"
 	"github.com/influxdata/flux/plan"
 	"github.com/influxdata/flux/values"
-	"github.com/influxdata/influxdb/v2/models"
-	"github.com/influxdata/influxdb/v2/query"
-	storage "github.com/influxdata/influxdb/v2/storage/reads"
-	"github.com/influxdata/influxdb/v2/storage/reads/datatypes"
-	"github.com/influxdata/influxdb/v2/tsdb/cursors"
+	"github.com/influxdata/influxdb/flux/stdlib/influxdata/influxdb"
+	"github.com/influxdata/influxdb/kit/platform/errors"
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/storage/reads/datatypes"
+	"github.com/influxdata/influxdb/tsdb/cursors"
 )
 
 // GroupCursorError is returned when two different cursor types
@@ -55,15 +53,15 @@ type storageTable interface {
 }
 
 type storeReader struct {
-	s storage.Store
+	s Store
 }
 
 // NewReader returns a new storageflux reader
-func NewReader(s storage.Store) query.StorageReader {
+func NewReader(s Store) influxdb.Reader {
 	return &storeReader{s: s}
 }
 
-func (r *storeReader) ReadFilter(ctx context.Context, spec query.ReadFilterSpec, alloc *memory.Allocator) (query.TableIterator, error) {
+func (r *storeReader) ReadFilter(ctx context.Context, spec influxdb.ReadFilterSpec, alloc *memory.Allocator) (influxdb.TableIterator, error) {
 	return &filterIterator{
 		ctx:   ctx,
 		s:     r.s,
@@ -73,7 +71,7 @@ func (r *storeReader) ReadFilter(ctx context.Context, spec query.ReadFilterSpec,
 	}, nil
 }
 
-func (r *storeReader) ReadGroup(ctx context.Context, spec query.ReadGroupSpec, alloc *memory.Allocator) (query.TableIterator, error) {
+func (r *storeReader) ReadGroup(ctx context.Context, spec influxdb.ReadGroupSpec, alloc *memory.Allocator) (influxdb.TableIterator, error) {
 	return &groupIterator{
 		ctx:   ctx,
 		s:     r.s,
@@ -83,7 +81,7 @@ func (r *storeReader) ReadGroup(ctx context.Context, spec query.ReadGroupSpec, a
 	}, nil
 }
 
-func (r *storeReader) ReadWindowAggregate(ctx context.Context, spec query.ReadWindowAggregateSpec, alloc *memory.Allocator) (query.TableIterator, error) {
+func (r *storeReader) ReadWindowAggregate(ctx context.Context, spec influxdb.ReadWindowAggregateSpec, alloc *memory.Allocator) (influxdb.TableIterator, error) {
 	return &windowAggregateIterator{
 		ctx:   ctx,
 		s:     r.s,
@@ -93,7 +91,7 @@ func (r *storeReader) ReadWindowAggregate(ctx context.Context, spec query.ReadWi
 	}, nil
 }
 
-func (r *storeReader) ReadTagKeys(ctx context.Context, spec query.ReadTagKeysSpec, alloc *memory.Allocator) (query.TableIterator, error) {
+func (r *storeReader) ReadTagKeys(ctx context.Context, spec influxdb.ReadTagKeysSpec, alloc *memory.Allocator) (influxdb.TableIterator, error) {
 	return &tagKeysIterator{
 		ctx:       ctx,
 		bounds:    spec.Bounds,
@@ -104,7 +102,7 @@ func (r *storeReader) ReadTagKeys(ctx context.Context, spec query.ReadTagKeysSpe
 	}, nil
 }
 
-func (r *storeReader) ReadTagValues(ctx context.Context, spec query.ReadTagValuesSpec, alloc *memory.Allocator) (query.TableIterator, error) {
+func (r *storeReader) ReadTagValues(ctx context.Context, spec influxdb.ReadTagValuesSpec, alloc *memory.Allocator) (influxdb.TableIterator, error) {
 	return &tagValuesIterator{
 		ctx:       ctx,
 		bounds:    spec.Bounds,
@@ -119,8 +117,8 @@ func (r *storeReader) Close() {}
 
 type filterIterator struct {
 	ctx   context.Context
-	s     storage.Store
-	spec  query.ReadFilterSpec
+	s     Store
+	spec  influxdb.ReadFilterSpec
 	stats cursors.CursorStats
 	cache *tagsCache
 	alloc *memory.Allocator
@@ -130,8 +128,8 @@ func (fi *filterIterator) Statistics() cursors.CursorStats { return fi.stats }
 
 func (fi *filterIterator) Do(f func(flux.Table) error) error {
 	src := fi.s.GetSource(
-		uint64(fi.spec.OrganizationID),
-		uint64(fi.spec.BucketID),
+		fi.spec.Database,
+		fi.spec.RetentionPolicy,
 	)
 
 	// Setup read request
@@ -158,7 +156,7 @@ func (fi *filterIterator) Do(f func(flux.Table) error) error {
 	return fi.handleRead(f, rs)
 }
 
-func (fi *filterIterator) handleRead(f func(flux.Table) error, rs storage.ResultSet) error {
+func (fi *filterIterator) handleRead(f func(flux.Table) error, rs ResultSet) error {
 	// these resources must be closed if not nil on return
 	var (
 		cur   cursors.Cursor
@@ -234,8 +232,8 @@ READ:
 
 type groupIterator struct {
 	ctx   context.Context
-	s     storage.Store
-	spec  query.ReadGroupSpec
+	s     Store
+	spec  influxdb.ReadGroupSpec
 	stats cursors.CursorStats
 	cache *tagsCache
 	alloc *memory.Allocator
@@ -245,8 +243,8 @@ func (gi *groupIterator) Statistics() cursors.CursorStats { return gi.stats }
 
 func (gi *groupIterator) Do(f func(flux.Table) error) error {
 	src := gi.s.GetSource(
-		uint64(gi.spec.OrganizationID),
-		uint64(gi.spec.BucketID),
+		gi.spec.Database,
+		gi.spec.RetentionPolicy,
 	)
 
 	// Setup read request
@@ -261,7 +259,7 @@ func (gi *groupIterator) Do(f func(flux.Table) error) error {
 	req.Range.Start = int64(gi.spec.Bounds.Start)
 	req.Range.End = int64(gi.spec.Bounds.Stop)
 
-	if len(gi.spec.GroupKeys) > 0 && gi.spec.GroupMode == query.GroupModeNone {
+	if len(gi.spec.GroupKeys) > 0 && gi.spec.GroupMode == influxdb.GroupModeNone {
 		return &errors.Error{
 			Code: errors.EInternal,
 			Msg:  "cannot have group mode none with group key values",
@@ -287,10 +285,10 @@ func (gi *groupIterator) Do(f func(flux.Table) error) error {
 	return gi.handleRead(f, rs)
 }
 
-func (gi *groupIterator) handleRead(f func(flux.Table) error, rs storage.GroupResultSet) error {
+func (gi *groupIterator) handleRead(f func(flux.Table) error, rs GroupResultSet) error {
 	// these resources must be closed if not nil on return
 	var (
-		gc    storage.GroupCursor
+		gc    GroupCursor
 		cur   cursors.Cursor
 		table storageTable
 	)
@@ -386,11 +384,11 @@ func determineAggregateMethod(agg string) (datatypes.Aggregate_AggregateType, er
 	return 0, fmt.Errorf("unknown aggregate type %q", agg)
 }
 
-func convertGroupMode(m query.GroupMode) datatypes.ReadGroupRequest_Group {
+func convertGroupMode(m influxdb.GroupMode) datatypes.ReadGroupRequest_Group {
 	switch m {
-	case query.GroupModeNone:
+	case influxdb.GroupModeNone:
 		return datatypes.GroupNone
-	case query.GroupModeBy:
+	case influxdb.GroupModeBy:
 		return datatypes.GroupBy
 	}
 	panic(fmt.Sprint("invalid group mode: ", m))
@@ -577,7 +575,7 @@ func determineTableColsForGroup(tagKeys [][]byte, typ flux.ColType, agg *datatyp
 	return cols, defs
 }
 
-func groupKeyForGroup(kv [][]byte, spec *query.ReadGroupSpec, bnds execute.Bounds) flux.GroupKey {
+func groupKeyForGroup(kv [][]byte, spec *influxdb.ReadGroupSpec, bnds execute.Bounds) flux.GroupKey {
 	cols := make([]flux.ColMeta, 2, len(spec.GroupKeys)+2)
 	vs := make([]values.Value, 2, len(spec.GroupKeys)+2)
 	cols[startColIdx] = flux.ColMeta{
@@ -605,8 +603,8 @@ func groupKeyForGroup(kv [][]byte, spec *query.ReadGroupSpec, bnds execute.Bound
 
 type windowAggregateIterator struct {
 	ctx   context.Context
-	s     storage.Store
-	spec  query.ReadWindowAggregateSpec
+	s     Store
+	spec  influxdb.ReadWindowAggregateSpec
 	stats cursors.CursorStats
 	cache *tagsCache
 	alloc *memory.Allocator
@@ -616,8 +614,8 @@ func (wai *windowAggregateIterator) Statistics() cursors.CursorStats { return wa
 
 func (wai *windowAggregateIterator) Do(f func(flux.Table) error) error {
 	src := wai.s.GetSource(
-		uint64(wai.spec.OrganizationID),
-		uint64(wai.spec.BucketID),
+		wai.spec.Database,
+		wai.spec.RetentionPolicy,
 	)
 
 	// Setup read request
@@ -681,7 +679,7 @@ func isSelector(kind plan.ProcedureKind) bool {
 	return kind == FirstKind || kind == LastKind || kind == MinKind || kind == MaxKind
 }
 
-func (wai *windowAggregateIterator) handleRead(f func(flux.Table) error, rs storage.ResultSet) error {
+func (wai *windowAggregateIterator) handleRead(f func(flux.Table) error, rs ResultSet) error {
 	createEmpty := wai.spec.CreateEmpty
 
 	selector := len(wai.spec.Aggregates) > 0 && isSelector(wai.spec.Aggregates[0])
@@ -839,16 +837,16 @@ func isAggregateCount(kind plan.ProcedureKind) bool {
 type tagKeysIterator struct {
 	ctx       context.Context
 	bounds    execute.Bounds
-	s         storage.Store
-	readSpec  query.ReadTagKeysSpec
+	s         Store
+	readSpec  influxdb.ReadTagKeysSpec
 	predicate *datatypes.Predicate
 	alloc     *memory.Allocator
 }
 
 func (ti *tagKeysIterator) Do(f func(flux.Table) error) error {
 	src := ti.s.GetSource(
-		uint64(ti.readSpec.OrganizationID),
-		uint64(ti.readSpec.BucketID),
+		ti.readSpec.Database,
+		ti.readSpec.RetentionPolicy,
 	)
 
 	var req datatypes.TagKeysRequest
@@ -922,16 +920,16 @@ func (ti *tagKeysIterator) Statistics() cursors.CursorStats {
 type tagValuesIterator struct {
 	ctx       context.Context
 	bounds    execute.Bounds
-	s         storage.Store
-	readSpec  query.ReadTagValuesSpec
+	s         Store
+	readSpec  influxdb.ReadTagValuesSpec
 	predicate *datatypes.Predicate
 	alloc     *memory.Allocator
 }
 
 func (ti *tagValuesIterator) Do(f func(flux.Table) error) error {
 	src := ti.s.GetSource(
-		uint64(ti.readSpec.OrganizationID),
-		uint64(ti.readSpec.BucketID),
+		ti.readSpec.Database,
+		ti.readSpec.RetentionPolicy,
 	)
 
 	var req datatypes.TagValuesRequest
```

Reply:
Might be worth renaming the `influxdb` import to `query` for easier backporting

Reply:
I'm going to hold off for now and do all the renaming / rearranging as part of #21112